proxy/envoy/config.go

```diff
@@ -158,62 +162,88 @@ func buildSidecar(env proxy.Environment, sidecar proxy.Node) (Listeners, Cluster
-
-	listeners := append(inbound, outbound...)
-	clusters := append(inClusters, outClusters...)
-
-	// If management listener port and service port are same, bad things happen
-	// when running in kubernetes, as the probes stop responding. So, append
-	// non overlapping listeners only.
-	for i := range mgmtListeners {
-		m := mgmtListeners[i]
-		c := mgmtClusters[i]
-		l := listeners.GetByAddress(m.Address)
-		if l != nil {
-			glog.Warningf("Omitting listener for management address %s (%s) due to collision with service listener %s (%s)",
-				m.Name, m.Address, l.Name, l.Address)
-			continue
+	listeners := make(Listeners, 0)
+	clusters := make(Clusters, 0)
+
+	if env.Mesh.ProxyListenPort > 0 {
+		inbound, inClusters := buildInboundListeners(env.Mesh, sidecar, instances)
+		outbound, outClusters := buildOutboundListeners(env.Mesh, sidecar, instances, services, env.IstioConfigStore)
+		mgmtListeners, mgmtClusters := buildMgmtPortListeners(env.Mesh, managementPorts, sidecar.IPAddress)
+
+		listeners = append(listeners, inbound...)
+		listeners = append(listeners, outbound...)
+		clusters = append(clusters, inClusters...)
+		clusters = append(clusters, outClusters...)
+
+		// If management listener port and service port are same, bad things happen
+		// when running in kubernetes, as the probes stop responding. So, append
+		// non overlapping listeners only.
+		for i := range mgmtListeners {
+			m := mgmtListeners[i]
+			c := mgmtClusters[i]
+			l := listeners.GetByAddress(m.Address)
+			if l != nil {
+				glog.Warningf("Omitting listener for management address %s (%s) due to collision with service listener %s (%s)",
+					m.Name, m.Address, l.Name, l.Address)
+				continue
+			}
+			listeners = append(listeners, m)
+			clusters = append(clusters, c)
+		}
+
+		// set bind to port values for port redirection
+		for _, listener := range listeners {
+			listener.BindToPort = false
 		}
-		listeners = append(listeners, m)
-		clusters = append(clusters, c)
-	}
 
-	// set bind to port values for port redirection
-	for _, listener := range listeners {
-		listener.BindToPort = false
+		// add an extra listener that binds to the port that is the recipient of the iptables redirect
+		listeners = append(listeners, &Listener{
+			Name:           VirtualListenerName,
+			Address:        fmt.Sprintf("tcp://%s:%d", WildcardAddress, env.Mesh.ProxyListenPort),
+			BindToPort:     true,
+			UseOriginalDst: true,
+			Filters:        make([]*NetworkFilter, 0),
+		})
 	}
 
-	// add an extra listener that binds to the port that is the recipient of the iptables redirect
-	listeners = append(listeners, &Listener{
-		Name:           VirtualListenerName,
-		Address:        fmt.Sprintf("tcp://%s:%d", WildcardAddress, env.Mesh.ProxyListenPort),
-		BindToPort:     true,
-		UseOriginalDst: true,
-		Filters:        make([]*NetworkFilter, 0),
-	})
+	// enable HTTP PROXY port if necessary; this will add an RDS route for this port
+	if env.Mesh.ProxyHttpPort > 0 {
```

Comment:
This should be else. Not sure what value one gets by having virtual and physical (non transparent) listeners in same config. Also, let's get rid of the http proxy port option. We can have a single proxy listen port and have a proxy mode flag that is transparent or non transparent. The http proxy should be a Boolean (allowing people to decide if they want to enable http_proxy and https_proxy) or not. (I am sorry I couldn't give this feedback on the istio/API pr for this option, as it was merged before I got a change to review).

This way, it's a cleaner config file in the end. We also need a inbound route. And since we are going down this route, for sanity sake, we need to use the host header to route the request to local cluster or the remote vm node. That means using source name. Alternatively, we create two listeners - ingress and egress listener like lyft. That would require the end user to register the vm with the ingress listener in the service registry. 

This is a very important change and will affect how things are being done in vm only deployments like lyft. So let's do it in the right way.